src/core/shape_inference/include/range_shape_inference.hpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/core/validation_util.hpp>
+#include <openvino/op/range.hpp>
+
+#include "utils.hpp"
+namespace ov {
+namespace op {
+namespace v4 {
+
+template <class T>
+inline bool get_data_as_double(
+    size_t idx,
+    const ov::Node* op,
+    std::vector<double>& axes_value,
+    const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    if (constant_data.count(idx)) {
+        axes_value = ov::opset1::Constant(constant_data.at(idx)).cast_vector<double>();
+    } else {
+        const auto& constant = ov::as_type_ptr<ov::opset1::Constant>(op->get_input_node_shared_ptr(idx));
+        NODE_VALIDATION_CHECK(op, constant != nullptr, "Static shape inference lacks constant data on port ", idx);
+        axes_value = constant->cast_vector<double>();
+    }
+    return true;
+}
+
+template <>
+inline bool get_data_as_double<ov::PartialShape>(
+    size_t idx,
+    const ov::Node* op,
+    std::vector<double>& axes_value,
+    const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data) {
+    if (constant_data.count(idx)) {
+        axes_value = ov::opset1::Constant(constant_data.at(idx)).cast_vector<double>();
+    } else if (const auto& constant = ov::get_constant_from_source(op->input_value(idx))) {
+        axes_value = constant->cast_vector<double>();
+    } else {
+        return false;
+    }
+    return true;
+}
+
+template <class T>
+void shape_infer(const Range* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data) {
```

Comment:
Please set default value for the `constant_data` parameter

Reply:
OK